bagua/torch_api/data_parallel/distributed.py

```diff
@@ -0,0 +1,316 @@
+                       bucket can potentially overlap with backward computation.
+                       :attr:`bucket_cap_mb` controls the bucket size in
+                       MegaBytes (MB). (default: 25)
+        find_unused_parameters (bool, optional): Traverse the autograd graph from all
+                               tensors contained in the return value of the
+                               wrapped module's ``forward`` function. Parameters
+                               that don't receive gradients as part of this
+                               graph are preemptively marked as being ready to
+                               be reduced. In addition, parameters that may have
+                               been used in the wrapped module's ``forward``
+                               function but were not part of loss computation and
+                               thus would also not receive gradients are
+                               preemptively marked as ready to be reduced.
+                               (default: ``False``)
+        check_reduction (bool, optional): This argument is deprecated.
+        gradient_as_bucket_view (bool, optional): When set to ``True``, gradients will be views
+                      pointing to different offsets of ``allreduce`` communication
+                      buckets. This can reduce peak memory usage, where the
+                      saved memory size will be equal to the total gradients
+                      size. Moreover, it avoids the overhead of copying between
+                      gradients and ``allreduce`` communication buckets. When
+                      gradients are views, ``detach_()`` cannot be called on the
+                      gradients. If hitting such errors, please fix it by
+                      referring to the :meth:`~torch.optim.Optimizer.zero_grad`
+                      function in ``torch/optim/optimizer.py`` as a solution.
+        optimizers (List[torch.optim.Optimizer], optional): Defaults to [].
+        algorithm (bagua.torch_api.algorithms.Algorithm, optional): Data
+                parallel distributed algorithm, decide how to communication mode
+                and the way the model is updated. Defaults to GradientAllReduceAlgorithm().
+
+    Returns:
+        Union[TorchDistributedDataParallel, DistributedDataParallel_V1_9_0]: parallelized module
+
+    Example::
+
+        >>> bagua.init_process_group()
+        >>> net = bagua.data_parallel.DistributedDataParallel(model, pg)
+
+    """
+
+    check_list = [
+        device_ids is None,
+        output_device is None,
+        dim == 0,
+        broadcast_buffers is True,
+        find_unused_parameters is False,
+        check_reduction is False,
+    ]
+    if not all(check_list):
+        warnings.warn("Input parameters are not supported yet, return torch native DistributedDataParallel.")
+        return TorchDistributedDataParallel(
```

Comment:
warn("Some parameters passed into BaguaDistributedDataParallel have not been supported yet. Bagua has automatically fallback to upstream PyTorch DistributedDataParallel implementation. If this is unexpected, please submit an issue to xxxxxxxx (our github url). Thanks.")